pjsip/include/pjsip-ua/sip_siprec.h

```diff
@@ -0,0 +1,116 @@
+#ifndef __PJSIP_SIPREC_H__
+#define __PJSIP_SIPREC_H__
+
+/**
+ * @file sip_siprec.h
+ * @brief SIP Session Recording Protocol (siprec) support (RFC 7866 - Session Recording Protocol in SIP)
+ */
+
+
+#include <pjsip-ua/sip_inv.h>
+#include <pjsip/sip_msg.h>
+
+/**
+ * @defgroup PJSIP_SIPREC SIP Session Recording Protocol (siprec) support (RFC 7866 - Session Recording Protocol in SIP)
+ * @brief SIP Session Recording Protocol support (RFC 7866 - Session Recording Protocol in SIP)
+ * @{
+ *
+ * \section PJSIP_SIPREC_REFERENCE References
+ *
+ * References:
+ *  - <A HREF="http://www.ietf.org/rfc/rfc7866.txt">RFC 7866: Session Recording Protocol (siprec)
+ *    in the Session Initiation Protocol (SIP)</A>
+ */
+PJ_BEGIN_DECL
+
+/**
+ * Initialize siprec module. This function must be called once during
+ * application initialization, to register siprec module to SIP endpoint.
+ *
+ * @param endpt         The SIP endpoint instance.
+ *
+ * @return              PJ_SUCCESS if module is successfully initialized.
+ */
+PJ_DECL(pj_status_t) pjsip_siprec_init_module(pjsip_endpoint *endpt);
+
+
+/**
+ * Check if the value of Require header is equal to siprec.
+ * 
+ * @param req_hdr         Require header.
+ * 
+ * @return                PJ_TRUE if value of Require header is equal to siprec.
+ */
+PJ_DECL(pj_status_t) pjsip_siprec_verify_require_hdr(pjsip_require_hdr *req_hdr);
+
+
+/**
+ * Checks if there is an attribute label for each media in the SDP.
+ * 
+ * @param media          The SDP media.
+ * 
+ * @return               PJ_TRUE if a label exists in the SDP.
+ */
+PJ_DECL(pj_status_t) pjsip_siprec_verify_sdp_attr_label(pjmedia_sdp_session *sdp);
```

Comment:
Well, this is still strange :)
Why not just put the implementation inline?

Reply:
You're right, I fixed it.
Do you think I should define `STR_LABEL` inside the function or at the top of the header file?
One more thing, what do you think about making the `pjsip_siprec_verify_require_hdr` function inline?

Reply:
Did I understand your point correctly?

Or are you suggesting that instead of creating the `pjsip_siprec_verify_sdp_attr_label` function, I should place it directly at `pjsip_siprec_verify_request`?

Reply:
Re: `STR_LABEL`, it should be in the `.c` file instead of header file. If it is used only once or twice, putting it in inside the function is okay, otherwise, better in the (top of) file.

Re: `pjsip_siprec_verify_require_hdr()`, it is okay inline or not.

Re: `pjsip_siprec_verify_sdp_attr_label()`, this is more like an SDP API, so better the latter approach, remove the API & place the implementation directly inside `pjsip_siprec_verify_request()`.

Reply:
fix it.

Reply:
The review of labels is a more general topic than SIPREC. In other words, when we have multiple media, there must always be a label, and later on, I will remove this review from SIPREC and place it in a more general section.